packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -424,7 +424,7 @@ class DevFS {
                          fileFilter: fileFilter);
     if (fs.isFileSync(_packagesFilePath)) {
       printTrace('Scanning package files');
-      await _scanPackages(fileFilter);
+      await _scanPackages(fileFilter, generator != null);
```

Comment:
Can you add a comment here that we detect we are in previewDart2 mode by generator (should have been named compiler?? is present) or have a local
bool previewDart2 = (generator != null);
to make that more obvious.

Reply:
Done